### PR TITLE
Feature/login models

### PR DESCRIPTION
**Pull request by LeandroSoria2**
### What changes does this PR include?
**Description**
Add devise gem in the login
**Summary**
- add gem devise
- add user model and migration. 
- add codeowners file

**Trello card**
[Trello card](https://trello.com/c/Fj0j3D5I)

### CODE REVIEW

File: Gemfile
Comment (JulianPasquale) [inline Gemfile:6]: I think we don't need this empty line here 🙂

File: CODEOWNERS
Comment (JulianPasquale) [inline CODEOWNERS:8]: You might have to grant write access to us to make this `CODEOWNERS` file valid

File: config/routes.rb
Comment (JulianPasquale) [inline config/routes.rb:6]: What is this? I think this should not be here

File: config/routes.rb
Comment (JulianPasquale) [inline config/routes.rb:7]: Don't we have to also add device routes to this file?

File: Gemfile
Comment (JulianPasquale) [inline Gemfile:45]: Is this going to be a full-stack app (with Rails views) or just an API?

If it's an API, you might have to consider using another gem that provides auth tokens instead of storing session in cookies. I think the one we use more is https://github.com/lynndylanhurley/devise_token_auth

File: db/schema.rb
Comment (diego-suarez) [inline db/schema.rb:17]: Looks like you created the users table, but no matching model on `app/models/user.rb`. maybe you missed that file?

File: CODEOWNERS
Comment (LeandroSoria2) [reply CODEOWNERS:8]: @JulianPasquale I sent the invitation for colaboration.

File: db/schema.rb
Comment (LeandroSoria2) [reply db/schema.rb:17]: @diego-suarez I Changed the gem in the last commit.

File: Gemfile
Comment (diego-suarez) [inline Gemfile:45]: Are you explicity setting the git repo for a particular reason? if not, maybe you can remove the git reference so it will be fetched from rubygems.org as all other gems

File: config/routes.rb
Comment (diego-suarez) [inline config/routes.rb:2]: wdyt about adding some api request tests? like with a valid user and password, an invalid user, invalid password, etc. and expect for the desired responses.
Btw I was looking at the project setup and looks like you did not created the project with `RSpec` as the testing tool, and by default it uses `minitest`
I don't know if there is another task for setting up the testing tool (or is out of the scope of the target project), but just to mention we commonly use rspec instead of minitest, and personally I would recommend you too 😅
https://github.com/rootstrap/tech-guides/tree/master/ruby/rspec

Comment (diego-suarez) [inline config/routes.rb:2]: wdyt about adding some api request tests? like with a valid user and password, an invalid user, invalid password, etc. and expect for the desired responses.
Btw I was looking at the project setup and looks like you did not created the project with `RSpec` as the testing tool, and by default it uses `minitest`
I don't know if there is another task for setting up the testing tool (or is out of the scope of the target project), but just to mention we commonly use rspec instead of minitest, and personally I would recommend you too 😅
https://github.com/rootstrap/tech-guides/tree/master/ruby/rspec

**Review by diego-suarez — APPROVED** [review]
good job! left a comment related to testing but can be ignored if out of scope

File: app/controllers/application_controller.rb
Comment (LeandroSoria2) [reply app/controllers/application_controller.rb:2]: corrected in the next pr

File: db/migrate/20220518201152_devise_token_auth_create_users.rb
Comment (LeandroSoria2) [reply db/migrate/20220518201152_devise_token_auth_create_users.rb:24]: corrected in the next pr

Comment (JulianPasquale) [reply db/migrate/20220518201152_devise_token_auth_create_users.rb:24]: Can't we fix it here?

I'd suggest to not merge this migration if we know that we'll not need these columns. Also I don't see any other PR created, so I really encourage you to make all the changes here 🙂

File: db/migrate/20220518201152_devise_token_auth_create_users.rb
Comment (AgostinaDev) [inline db/migrate/20220518201152_devise_token_auth_create_users.rb:33]:
```suggestion
      t.string :last_name
```

**Review by JulianPasquale — APPROVED** [review]

**Pull request merged**
